### PR TITLE
fix(combobox): odd cursor behavior

### LINKED ISSUE #17222: [Bug]: Odd Cursor Behaviour While Using ComboBox Component
### Package

@carbon/react

### Browser

Chrome, Firefox

### Package version

v1.63.2

### React version

v18.3.1

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-combobox--default

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Application/PAL

IBM Data Replication

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by ariellalgilmore**
Closes #17222 

Odd cursor behavior in combobox when typing "abc" then moving cursor between "a" and "b" the text was appear as "a1bc23" instead of "a123bc".

https://github.com/user-attachments/assets/c5cde70e-b0a3-4ea7-8b06-f1683df4d49d

resources: https://github.com/downshift-js/downshift/blob/master/src/hooks/useCombobox/README.md?plain=1#L581-L611

#### Changelog

**Changed**

- removed cursor positioning and added setInputValue(newValue) on the change: 

#### Testing / Reviewing

Check combobox storybook

### CODE REVIEW

**Review by Gururajj77 — APPROVED** [review]
This looks good, thanks for refining this @ariellalgilmore

File: packages/react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -797,12 +795,6 @@ const ComboBox = forwardRef(
       downshiftSetInputValue,
       toggleMenu,
     ]);
-    useEffect(() => {
-      if (inputRef.current) {
-        inputRef.current.setSelectionRange(cursorPosition, cursorPosition);
-      }
-    }, [inputValue, cursorPosition]);
-
```
Comment (tay1orjones) [inline packages/react/src/components/ComboBox/ComboBox.tsx:805]: Any idea what this was used for originally? It's wild that this whole useEffect was in there that not only wasn't needed, but actively caused a bug!

**Review by tay1orjones — APPROVED** [review]

**Review by riddhybansal — COMMENTED** [review]
This LGTM !!

File: packages/react/src/components/ComboBox/ComboBox.tsx (hunk 1 above)
Comment (ariellalgilmore) [reply packages/react/src/components/ComboBox/ComboBox.tsx:805]: the cursorPosition came from the recent Autocomplete with Typeahead PR, but the odd cursor behavior was happening before that. When trying to fix the bug though, these additions seemed to probably help a weird cursor position when adding the autocomplete, so I realized fixing the original cursor bug also meant we could get rid of these new additions

**Pull request merged**